### PR TITLE
fix decoding IP based hostname

### LINKED ISSUE #10017: IP_ENCODED_AS_HOSTNAME with automatic TLS is throwing a warning
Below line is causing a WARNING message in Coordinator logs if node name is following the new IP address encoded value.

https://github.com/trinodb/trino/blob/e3cf28834f85bc1271311a265617fb273d7b34be/core/trino-main/src/main/java/io/trino/execution/scheduler/UniformNodeSelectorFactory.java#L144

Exception:
```
2021-11-19T04:48:29.162Z        WARN    Query-20211119_044828_00000_9hst5-310   io.trino.execution.scheduler.UniformNodeSelectorFactory Unable to resolve host name for node: InternalNode{nodeIdentifier=10-10-11-11, internalUri=https://10-10-11-11.ip:8443, nodeVersion=365_SNAPSHOT}
java.net.UnknownHostException: 10-10-11-11.ip: Name or service not known
        at java.base/java.net.Inet6AddressImpl.lookupAllHostAddr(Native Method)
        at java.base/java.net.InetAddress$PlatformNameService.lookupAllHostAddr(InetAddress.java:929)
        at java.base/java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1519)
        at java.base/java.net.InetAddress$NameServiceAddresses.get(InetAddress.java:848)
        at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1509)
        at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1368)
        at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1302)
        at java.base/java.net.InetAddress.getByName(InetAddress.java:1252)
        at io.trino.execution.scheduler.UniformNodeSelectorFactory.createNodeMap(UniformNodeSelectorFactory.java:144)
        at io.trino.execution.scheduler.UniformNodeSelectorFactory.lambda$createNodeSelector$0(UniformNodeSelectorFactory.java:109)
        at com.google.common.base.Suppliers$ExpiringMemoizingSupplier.get(Suppliers.java:243)
        at io.trino.execution.scheduler.UniformNodeSelector.selectRandomNodes(UniformNodeSelector.java:118)
        at io.trino.execution.scheduler.NodeSelector.selectRandomNodes(NodeSelector.java:34)
        at io.trino.sql.planner.SystemPartitioningHandle.getNodePartitionMap(SystemPartitioningHandle.java:147)
        at io.trino.sql.planner.NodePartitioningManager.getNodePartitioningMap(NodePartitioningManager.java:133)
        at io.trino.execution.scheduler.SqlQueryScheduler.lambda$new$1(SqlQueryScheduler.java:211)
        at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1133)
        at io.trino.execution.scheduler.SqlQueryScheduler.lambda$new$2(SqlQueryScheduler.java:211)
        at io.trino.execution.scheduler.SqlQueryScheduler.createStages(SqlQueryScheduler.java:456)
        at io.trino.execution.scheduler.SqlQueryScheduler.<init>(SqlQueryScheduler.java:203)
        at io.trino.execution.scheduler.SqlQueryScheduler.createSqlQueryScheduler(SqlQueryScheduler.java:147)
        at io.trino.execution.SqlQueryExecution.planDistribution(SqlQueryExecution.java:538)
        at io.trino.execution.SqlQueryExecution.start(SqlQueryExecution.java:420)
        at io.trino.execution.SqlQueryManager.createQuery(SqlQueryManager.java:237)
        at io.trino.dispatcher.LocalDispatchQuery.lambda$startExecution$7(LocalDispatchQuery.java:143)
        at io.trino.$gen.Trino_365_SNAPSHOT____20211119_044656_2.run(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
```

### PR DESCRIPTION
**Pull request by sajjoseph**
fixes #10017 

@dain, please review

### CODE REVIEW

**Comment by sajjoseph** [comment]
@cla-bot check

File: core/trino-main/src/main/java/io/trino/execution/scheduler/UniformNodeSelectorFactory.java [hunk 1]
```diff
@@ -138,20 +139,26 @@ private NodeMap createNodeMap(Optional<CatalogName> catalogName)
...
-                InetAddress host = InetAddress.getByName(node.getInternalUri().getHost());
-                byHost.put(host, node);
-            }
-            catch (UnknownHostException e) {
-                if (inaccessibleNodeLogCache.getIfPresent(node) == null) {
-                    inaccessibleNodeLogCache.put(node, true);
-                    LOG.warn(e, "Unable to resolve host name for node: %s", node);
-                }
-            }
+            byHostAndPort.put(node.getHostAndPort(), node);
```
Comment (dain) [inline core/trino-main/src/main/java/io/trino/execution/scheduler/UniformNodeSelectorFactory.java:142]: We need a similar change in TopologyAwareNodeSelectorFactory, so I suggest we add the following method to InternalNode (right below `getInternalUri`):
```java
public InetAddress getInternalAddress()
        throws UnknownHostException
{
    Optional<InetAddress> address = tryDecodeHostnameToAddress(internalUri.getHost());
    if (address.isPresent()) {
        return address.get();
    }
    return InetAddress.getByName(internalUri.getHost());
}
```
Then we only change one line in the code... something like simply becomes:
```java
for (InternalNode node : nodes) {
    try {
        byHostAndPort.put(node.getHostAndPort(), node);
        byHost.put(node.getInternalAddress(), node);
    }
    catch (UnknownHostException e) {
        if (inaccessibleNodeLogCache.getIfPresent(node) == null) {
            inaccessibleNodeLogCache.put(node, true);
            LOG.warn(e, "Unable to resolve host name for node: %s", node);
        }
    }
}
```

**Comment by sajjoseph** [comment]
Thanks, Dain for the suggestion.
I updated the PR.
Please review when you get a moment.

File: core/trino-main/src/main/java/io/trino/execution/scheduler/UniformNodeSelectorFactory.java (hunk 1 above)
Comment (sajjoseph) [reply core/trino-main/src/main/java/io/trino/execution/scheduler/UniformNodeSelectorFactory.java]: Changes incorporated.

**Comment by sajjoseph** [comment]
@dain - gentle ping. It will be great if this can be part of 365 release.

**Review by dain — APPROVED** [review]

**Comment by dain** [comment]
I got the tests to pass in the alternate PR (no idea why tests won't run here).  This looks good so merging

**Pull request merged**